core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -1273,3 +1282,32 @@ case class ExternalClusterMode(override val sc: SparkContext,
 
 class TableNotFoundException(message: String, cause: Option[Throwable] = None)
     extends AnalysisException(message) with Serializable
+
+/**
+ * Expression that returns the dsid of the server containing the row.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_() - Returns the dsid of the server containing the row.")
+case class DSID() extends LeafExpression {
+
```

Comment:
Move this out into a separate file say "io/snappydata/functions.scala" so that we can add all our functions in one place.